lib/puppet/util/metric.rb

```diff
@@ -4,17 +4,24 @@
 
 # A class for handling metrics.  This is currently ridiculously hackish.
 class Puppet::Util::Metric
+  include Puppet::Util::PsychSupport
   include Puppet::Network::FormatSupport
 
   attr_accessor :type, :name, :value, :label
   attr_writer :values
 
   def self.from_data_hash(data)
-    metric = new(data['name'], data['label'])
-    metric.values = data['values']
+    metric = allocate
+    metric.initialize_from_hash(data)
     metric
   end
 
+  def initialize_from_hash(data)
+    @name = data['name']
+    @label = data['label'] || self.class.labelize(@name)
```

Comment:
They could, but that would change API since the `Util::Metric` class is not marked as `@api private`. Also, if we consider that all data hashes passed to `initialize_from_hash` are now `Data`, then a `to_s` call there would be redundant.